demos/python_demos/speech_recognition_demo/ctcdecode-numpy/setup.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is based in part on setup.py from https://github.com/parlance/ctcdecode,
+# commit 431408f22d93ef5ebc4422995111bbb081b971a9 on Apr 4, 2020, 20:54:49 UTC+1.
+#
+import multiprocessing.pool
+import os
+
+from setuptools import setup, find_packages, distutils
+from distutils.command.build import build as Build
+
+this_file = os.path.dirname(__file__)
+
+
+# monkey-patch for parallel compilation
+# See: https://stackoverflow.com/a/13176803
+def parallelCCompile(self,
+                     sources,
+                     output_dir=None,
+                     macros=None,
+                     include_dirs=None,
+                     debug=0,
+                     extra_preargs=None,
+                     extra_postargs=None,
+                     depends=None):
+    # those lines are copied from distutils.ccompiler.CCompiler directly
+    macros, objects, extra_postargs, pp_opts, build = self._setup_compile(
+        output_dir, macros, include_dirs, sources, depends, extra_postargs)
+    cc_args = self._get_cc_args(pp_opts, debug, extra_preargs)
+
+    # parallel code
+    def _single_compile(obj):
+        try:
+            src, ext = build[obj]
+        except KeyError:
+            return
+        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+
+    # convert to list, imap is evaluated on-demand
+    thread_pool = multiprocessing.pool.ThreadPool(12)
+    list(thread_pool.imap(_single_compile, objects))
+    return objects
+
+
+# hack compile to support parallel compiling
+distutils.ccompiler.CCompiler.compile = parallelCCompile
```

Comment:
since this part need only for speed up compilation and contains hardcoded number of threads.
Can we try to build this module using cmake instead, possibly without multithreading compilation?

Reply:
I can just remove parallelCCompile now, since I removed OpenFst dependency. With OpenFst this helped to drop compilation time from ~5 min to 1.5 min. Now without OpenFst, it became 15 sec and 6 sec -- it's insignificant now.

parallelCCompile was inherited from Parlance ctcdecode.

Reply:
Removed parallelCCompile and added invocation of "pip install ." from CMakeLists.txt.

Reply:
I already removed this in the later commit. You can switch from "Changes from <number> commits" to "Changes from all commits" in github's "Files changed" view top-left, and search for filename to check.